pmpro-mailchimp.php

```diff
@@ -1073,24 +1074,25 @@ function pmpromc_queueUserToUnsubscribeFromLists($user_id) {
  */
 function pmpromc_processUnsubscriptions($param) {
 	global $pmpromc_users_to_unsubscribe;
-	
+
 	//anything to do?
 	if(empty($pmpromc_users_to_unsubscribe))
 		return $param;
-	
+
 	//unsubscribe
 	foreach($pmpromc_users_to_unsubscribe as $user_id) {
-		pmpromc_unsubscribeFromLists($user_id);		
+		pmpromc_unsubscribeFromLists($user_id);
 	}
-	
+
 	//unset so we don't do this twice by accident
 	unset($pmpromc_users_to_unsubscribe);
-		
+
 	//sometimes called in a filter and we need to pass this back
 	return $param;
 }
 add_action('template_redirect', 'pmpromc_processUnsubscriptions', 2);
 add_filter('wp_redirect', 'pmpromc_processUnsubscriptions', 100);
+add_action('pmpro_membership_post_membership_expiry', 'pmpromc_processUnsubscriptions');

```

Comment:
Thanks for the fix, really appreciate it.

However I think we should call `pmpromc_unsubscribeFromLists` instead of `pmpromc_processUnsubscriptions` because `pmpromc_processUnsubscriptions` is using a global variable `$pmpromc_users_to_unsubscribe` that may be empty during the expiry processing.

`pmpro_membership_post_membership_expiry` hook passes the user id which fit nicely to function `pmpromc_unsubscribeFromLists`

Reply:
The reason we need to use the $pmpro_users_to_unsubscribe global and pmpromc_processUnsubscriptions() function is for Multiple Memberships per User support. If a user has 2 membership levels that get expired, we want to figure out how their lists should change once and hit the MailChimp API as little as possible. If we instead cancel memberships after each level change, we often will remove a user from a list, then add them back to the list, then remove them again or try to add them again/etc.

Which actually points to some other changes needed in this PR I'm going to comment on below.